src/ngraph/pass/manager.cpp

```diff
@@ -61,17 +61,26 @@ void pass::Manager::run_passes(shared_ptr<Function> func, bool transitive)
 {
     bool profile_enabled = getenv("NGRAPH_PROFILE_PASS_ENABLE") != nullptr;
 
-    vector<shared_ptr<Function>> fs;
+    vector<std::pair<shared_ptr<Function>, bool>> fs;
```

Comment:
Sorry I missed the module pass in our discussion so I see why you need `f_array` for the module passes. But `fs` and `f_array` are sort of duplicate. Maybe it makes sense to keep the `bool` flags in a separate array. Do you think that works better?